### ISSUE TITLE
[question] How to write a test_package for a python_requires recipe?

### ISSUE THREAD

**Issue by phbasler** [conan-io/conan#12408]
I want to make use of the `python_requires` feature. To test it I wanted to make a test_package to test my base class but I encountered a problem on how to reference the currently being created base recipe.

If I use `python_requires = self.tested_reference_str` in my `test_package/conanfile.py` then I get the error 
```python
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "D:\git\vikaphb\conan-python-requires-example\test_package\conanfile.py", line 12, in <module>
    class TempTestConan(ConanFile):
  File "D:\git\vikaphb\conan-python-requires-example\test_package\conanfile.py", line 14, in TempTestConan
    python_requires = self.tested_reference_str
NameError: name 'self' is not defined
```

My current workaround is to modify my `set_version` function of the `conanfile.py` to 
```python
    def set_version(self):
        self.version = "0.1.3"
        f = open("./test_package/version.txt","w")
        f.write(self.version)
```

and in the `test_package/conanfile.py` I write
```python
def get_version():
    f = open("version.txt","r")
    version = f.read()
    return version

class TempTestConan(ConanFile):

    python_requires = "pyreq/" + get_version()
```

I have a MWE here: https://github.com/phbasler/conan-python-requires-example

This works, but is not nice. Is there a way to get the `self.tested_reference_str` working with `python_requires`?

- [x ] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Exploring possible approach in https://github.com/conan-io/conan/pull/12413

**Comment by czoido** [comment]
Closed by https://github.com/conan-io/conan/pull/12413 will be released in next beta: 2.0-beta5
Thanks a lot for reporting @phbasler

**Comment by phbasler** [comment]
@czoido Thanks! Will this also be in one of the 1.x releases or only a 2.x feature?

**Comment by czoido** [comment]
Hi @phbasler,
I'm afraid that this will not be available in Conan 1.X as it also relies on the _package_type_ information that is only available in 2.0
